lib/helpers/preset-announcement.js

```diff
@@ -2,36 +2,45 @@
 const logger = require('sonos-discovery/lib/helpers/logger');
 const isRadioOrLineIn = require('../helpers/is-radio-or-line-in');
 
-function saveAll(system) {
+function savePresetPlayers(system, pset) {
   const backupPresets = system.zones.map((zone) => {
     const coordinator = zone.coordinator;
     const state = coordinator.state;
-    const preset = {
-      players: [
-        { roomName: coordinator.roomName, volume: state.volume }
-      ],
-      state: state.playbackState,
-      uri: coordinator.avTransportUri,
-      metadata: coordinator.avTransportUriMetadata,
-      playMode: {
-        repeat: state.playMode.repeat
-      }
-    };
-
-    if (!isRadioOrLineIn(preset.uri)) {
-      preset.trackNo = state.trackNo;
-      preset.elapsedTime = state.elapsedTime;
-    }
+    for (var index = 0; index < pset.players.length; ++index) {
+			if (pset.players[index].roomName == coordinator.roomName) {
+        const preset = {
+          players: [
+            { roomName: coordinator.roomName, volume: state.volume }
+          ],
+          state: state.playbackState,
+          uri: coordinator.avTransportUri,
+          metadata: coordinator.avTransportUriMetadata,
+          playMode: {
+            repeat: state.playMode.repeat
+          }
+        };
 
-    zone.members.forEach(function (player) {
-      if (coordinator.uuid != player.uuid)
-        preset.players.push({ roomName: player.roomName, volume: player.state.volume });
-    });
+        if (!isRadioOrLineIn(preset.uri)) {
+          preset.trackNo = state.trackNo;
+          preset.elapsedTime = state.elapsedTime;
+        }
 
-    return preset;
+        zone.members.forEach(function (player) {
+          if (coordinator.uuid != player.uuid)
+            preset.players.push({ roomName: player.roomName, volume: player.state.volume });
+        });
 
+        return preset;
+      }
+    }
   });
 
+  Object.keys(backupPresets).forEach(key => {
+		if (backupPresets[key] === undefined) {
+			delete backupPresets[key];
+		}
+	});
+
```

Comment:
instead of mutating the array to remove "empty", you can add a `.filter(Boolean)` after the map() when creating backupPresets.

Reply:
sorry, I don't know how